components/script/dom/element.rs

```diff
@@ -780,81 +787,76 @@ impl Element {
-                        existing_declarations.insert(0, incoming_declaration);
                     }
                 }
-
                 return;
             }
 
-            let (important, normal) = if style_priority == StylePriority::Important {
-                (declarations, vec![])
+            let (normal_count, important_count) = if importance.important() {
+                (0, declarations.len() as u32)
             } else {
-                (vec![], declarations)
+                (declarations.len() as u32, 0)
             };
 
             *inline_declarations = Some(PropertyDeclarationBlock {
-                important: Arc::new(important),
-                normal: Arc::new(normal),
+                declarations: Arc::new(declarations.into_iter().map(|d| (d, importance)).collect()),
+                normal_count: normal_count,
+                important_count: important_count,
             });
         }
 
-        update(self, declarations, style_priority);
+        update(self, declarations, importance);
         self.sync_property_with_attrs_style();
     }
 
     pub fn set_inline_style_property_priority(&self,
                                               properties: &[&str],
-                                              style_priority: StylePriority) {
+                                              new_importance: Importance) {
         {
             let mut inline_declarations = self.style_attribute().borrow_mut();
             if let &mut Some(ref mut declarations) = &mut *inline_declarations {
-              let (from, to) = if style_priority == StylePriority::Important {
-                  (&mut declarations.normal, &mut declarations.important)
-              } else {
-                  (&mut declarations.important, &mut declarations.normal)
-              };
-
-              // Usually, the reference counts of `from` and `to` will be 1 here. But transitions
-              // could make them greater than that.
-              let from = Arc::make_mut(from);
-              let to = Arc::make_mut(to);
-              let mut new_from = Vec::new();
-              for declaration in from.drain(..) {
-                  let name = declaration.name();
-                  if properties.iter().any(|p| name == **p) {
-                      to.push(declaration)
-                  } else {
-                      new_from.push(declaration)
-                  }
-              }
-              mem::replace(from, new_from);
+                // Usually, the reference counts of `from` and `to` will be 1 here. But transitions
+                // could make them greater than that.
+                let declarations = Arc::make_mut(&mut declarations.declarations);
+                for &mut (ref declaration, ref mut importance) in declarations {
+                    if properties.iter().any(|p| declaration.name() == **p) {
+                        *importance = new_importance
```

Comment:
forgot to update the count in this one, that's why that test is failing, sorry for not catching it before.